cli/src/cores/models/crawled-asset.ts

```diff
@@ -42,37 +42,30 @@ export class CrawledAsset {
     this.fileExtension = path.extname(this.path);
     this.fileSize = stats.size;
 
-    let hasSidecar = true;
-
     // TODO: doesn't xmp replace the file extension? Will need investigation
     const sideCarPath = `${this.path}.xmp`;
     try {
       fs.accessSync(sideCarPath, fs.constants.R_OK);
-    } catch (error) {
-      // No sidecar file
-      hasSidecar = false;
-    }
-    if (hasSidecar) {
-      this.sidecarPath = `${this.path}.xmp`;
-      this.sidecarData = await fs.promises.readFile(this.sidecarPath);
-    }
+      this.sidecarData = await fs.promises.readFile(sideCarPath);
+      this.sidecarPath = sideCarPath;
+    } catch (error) {}
   }
 
   async delete(): Promise<void> {
     return fs.promises.unlink(this.path);
   }
 
-  public async hash(): Promise<string> {
-    const hash = crypto.createHash('sha1');
+  hashFunction = crypto.createHash('sha1');
 
-    const sha1 = (filePath: string) =>
-      new Promise<string>((resolve, reject) => {
-        const rs = fs.createReadStream(filePath);
-        rs.on('error', reject);
-        rs.on('data', (chunk) => hash.update(chunk));
-        rs.on('end', () => resolve(hash.digest('hex')));
-      });
+  sha1 = (filePath: string) =>
+    new Promise<string>((resolve, reject) => {
+      const rs = fs.createReadStream(filePath);
+      rs.on('error', reject);
+      rs.on('data', (chunk) => this.hashFunction.update(chunk));
+      rs.on('end', () => resolve(this.hashFunction.digest('hex')));
+    });
```

Comment:
Moving them here makes them class methods, which is OK... maybe something like this is better:

```typescript
const sha1 = (filePath: string) => {
    const hash = crypto.createHash('sha1');
    return new Promise<string>((resolve, reject) => {
      const rs = fs.createReadStream(filePath);
      rs.on('error', reject);
      rs.on('data', (chunk) => hash.update(chunk));
      rs.on('end', () => resolve(hash.digest('hex')));
    });
}
```
Or, if you want the method to be on the class still

```typescript
  private sha1(filePath: string) {
    const hash = crypto.createHash('sha1');
    return new Promise<string>((resolve, reject) => {
      const rs = fs.createReadStream(filePath);
      rs.on('error', reject);
      rs.on('data', (chunk) => hash.update(chunk));
      rs.on('end', () => resolve(hash.digest('hex')));
    });
  }
```

Reply:
OK, did as per your suggestion